ls1intum/Artemis
# `Lectures`: Fix lecture attachment updates with the same name

### Checklist
#### General


#### Server


### Motivation and Context
The editing of lecture attachment units was broken. If you uploaded a file and replaced it in the edit mode with a file with the same name, the file was not possible to be downloaded anymore.

### Description
This PR fixes this issue by adjusting an equality check of paths, doing a workaround for buggy FileUtils function and deleting the cache correctly.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Lecture

1. Log in to Artemis
2. Navigate to Lecture Units
3. Create Attachment Unit
4. Edit the Attachment Unit and upload a different file with the same filename as before
5. Save
6. Verify that the attachment ist still downloadable


### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Code Review
#### Manual Tests


### Test Coverage







### Screenshots
